samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -1194,14 +1194,20 @@ static bool is_substring(const char *substr, const char *str)
 
 static bool data_cb(struct bt_data *data, void *user_data)
 {
-	char *name = user_data;
+	bool *device_found = user_data;
+	char name[NAME_LEN] = {0};
 
 	switch (data->type) {
 	case BT_DATA_NAME_SHORTENED:
 	case BT_DATA_NAME_COMPLETE:
 	case BT_DATA_BROADCAST_NAME:
 		memcpy(name, data->data, MIN(data->data_len, NAME_LEN - 1));
-		return false;
+		if (!(is_substring(CONFIG_TARGET_BROADCAST_NAME, name))) {
```

Comment:
```suggestion
		if (is_substring(CONFIG_TARGET_BROADCAST_NAME, name)) {
```

Reply:
Sorry, was a bit quick on the push-button. Fixed